### ISSUE TITLE
iPlanner: Imperative Path Planning. (arXiv:2302.11434v1 [cs.RO])

### ISSUE THREAD

**Issue by CoffeeKumazaki** [CoffeeKumazaki/arXiv#24894]
iPlanner: Imperative Path Planning. (arXiv:2302.11434v1 [cs.RO])<br>
https://ift.tt/8dMb4PU<br>
<p>The problem of path planning has been studied for years. Classic planning pipelines, including perception, mapping, and path searching, can result in latency and compounding errors between modules. While recent studies have demonstrated the effectiveness of end-to-end learning methods in achieving high planning efficiency, these methods often struggle to match the generalization abilities of classic approaches in handling different environments. Moreover, end-to-end training of policies often requires a large number of labeled data or training iterations to reach convergence. In this paper, we present a novel Imperative Learning (IL) approach. This approach leverages a differentiable cost map to provide implicit supervision during policy training, eliminating the need for demonstrations or labeled trajectories. Furthermore, the policy training adopts a Bi-Level Optimization (BLO) process, which combines network update and metric-based trajectory optimization, to generate a smooth and collision-free path toward the goal based on a single depth measurement. The proposed method allows the costs of predicted trajectories and task-level loss to be backpropagated through all layers to update the network with direct gradients. In our experiments, the method demonstrates around 4x faster planning than the classic approach and robustness against localization noise. Additionally, the IL approach enables the planner to generalize to various unseen environments, resulting in an overall 26-87% improvement in performance compared to baseline learning methods.</p>
<br>
<br>
via cs.RO updates on arXiv.org http://arxiv.org/<br>
